src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -54,45 +57,72 @@ g.test('mapAsync')
     if (oom) {
       // Should also reject in addition to the validation error.
       t.shouldReject('OperationError', promise);
+
+      // Should throw an OperationError because the buffer is not mapped.
+      // Note: not a RangeError because the state of the buffer is checked first.
+      t.shouldThrow('OperationError', () => {
+        buffer.getMappedRange();
+      });
+
+      // Should throw an OperationError because the buffer is already unmapped.
+      t.shouldThrow('OperationError', () => {
+        buffer.unmap();
+      });
     } else {
       await promise;
       const arraybuffer = buffer.getMappedRange();
       t.expect(arraybuffer.byteLength === size);
       buffer.unmap();
-      t.expect(arraybuffer.byteLength === 0);
+      t.expect(arraybuffer.byteLength === 0, 'Mapping should be detached');
     }
   });
 
 g.test('mappedAtCreation')
-  .params(
-    params()
-      .combine(pbool('oom')) //
-      .combine(poptions('usage', kBufferUsages))
+  .desc(
+    `Test creating a very large buffer mappedAtCreation buffer should produce
+an out-of-memory error if allocation fails.
+  - Because the buffer can be immediately mapped, getMappedRange does not throw an OperationError. It throws a RangeError because such a
+    large ArrayBuffer cannot be created.
+  - unmap() should not throw.
+  `
   )
+  .cases(oomAndSizeParams)
+  .subcases(() => params().combine(poptions('usage', kBufferUsages)))
   .fn(async t => {
-    const { oom, usage } = t.params;
-    const size = oom ? MAX_ALIGNED_SAFE_INTEGER : 16;
+    const { oom, usage, size } = t.params;
 
     const buffer = t.expectGPUError(
       'out-of-memory',
       () => t.device.createBuffer({ mappedAtCreation: true, size, usage }),
       oom
     );
 
-    const f = () => buffer.getMappedRange(0, size);
+    const f = () => buffer.getMappedRange();
 
+    let mapping: ArrayBuffer | undefined = undefined;
     if (oom) {
       t.shouldThrow('RangeError', f);
     } else {
-      f();
+      mapping = f();
+    }
+    buffer.unmap();
+    if (mapping !== undefined) {
+      t.expect(mapping.byteLength === 0, 'Mapping should be detached');
     }
   });
 
 g.test('mappedAtCreation,smaller_getMappedRange')
-  .params(poptions('usage', kBufferUsages))
+  .desc(
+    `Test creating a very large buffer mappedAtCreation buffer should produce
+an out-of-memory error if allocation fails.
+  - Because the buffer can be immediately mapped, getMappedRange does not throw an OperationError. Calling it on a small range of the buffer succesfully returns an ArrayBuffer.
```

Comment:
```suggestion
  - Because the buffer can be immediately mapped, getMappedRange does not throw an OperationError. Calling it on a small range of the buffer successfully returns an ArrayBuffer.
```